tv_apps/device-deck/test/unit/bluetooth_manager_test.js

```diff
@@ -0,0 +1,190 @@
+/* global MockMozBluetooth, MockBTAdapter, BluetoothManager, MockEventTarget,
+          MockDiscoveryHandle */
+'use strict';
+
+require('/bower_components/evt/index.js');
+require('/shared/test/unit/mocks/mock_event_target.js');
+require('/shared/test/unit/mocks/mock_navigator_moz_bluetooth_v2.js');
+require('/js/bluetooth_loader.js');
+require('/js/bluetooth_manager.js');
+
+suite('device-deck/BluetoothManager', function() {
+
+  var realMozBluetooth;
+  var bluetoothManager;
+
+  suiteSetup(function() {
+    var MockDiscoveryHandle = function() {
+      this.ondevicefound = null;
+    };
+    MockDiscoveryHandle.prototype = Object.create(MockEventTarget.prototype);
+    window.MockDiscoveryHandle = MockDiscoveryHandle;
+
+    realMozBluetooth = navigator.mozBluetooth;
+    switchReadOnlyProperty(navigator, 'mozBluetooth', MockMozBluetooth);
+  });
+
+  suiteTeardown(function() {
+    window.MockDiscoveryHandle = undefined;
+    switchReadOnlyProperty(navigator, 'mozBluetooth', realMozBluetooth);
+  });
+
+  var switchReadOnlyProperty = function (originObject, propName, targetObj) {
+    Object.defineProperty(originObject, propName, {
+      configurable: true,
+      get: function() { return targetObj; }
+    });
+  };
+
+  var setupForMockDefaultAdapter = function(context) {
+    var mockDiscoveryHandle = new MockDiscoveryHandle();
+    MockBTAdapter.state = 'enabled';
+    MockBTAdapter.startDiscovery = function() {};
+    MockBTAdapter.stopDiscovery = function() {};
```

Comment:
Would it be simpler if we just write
`context.sinon.stub(MockBTAdapter, stopDiscovery, function() { ... }`?